### PR TITLE
feat(ActionRequirement): improve accessibility

### PR DESCRIPTION
**Pull request by mavericardo**
[JIRA Issue](https://gympass.atlassian.net/browse/AC-5254)

## Description 📄

- Add `aria-hidden` attr in the illustration;
- Add `as` attr with `h1` value and `aria-label`  in title components;
- Add `aria-label` in description text
- Add tabIndex in all elements from ActionRequirement

[Enter the description of your changes]

## Platforms 📲

- [ ] Web
- [ ] Mobile

## Type of change 🔍

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? 🧪

[Enter the tips to test this PR]

- [ ] Unit Test
- [ ] Snapshot Test

## Checklist: 🔍

- [ ] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [ ] Layout matches design prototype: [FIGMA](https://figma.com/file/YOUR_LINK)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Screenshots 📸

https://www.figma.com/design/8SVndQjnDLmApd3NntMIPQ/Accessibility?node-id=3314-55285&node-type=instance&t=fnXh0R28TsoWG9PT-0

Screenshot 2024-11-19 at 12 07 14

[Upload your screenshots here]

| Before | After |
| ------ | ----- |
|        |       |

### CODE REVIEW

File: packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx [hunk 1]
```diff
@@ -73,15 +73,15 @@ function ActionRequirement(props) {
   defineCompoundComponents();
   return (
     <StyledActionRequirement {...props}>
-      {illustration && <BoxIllustration>{illustration}</BoxIllustration>}
+      {illustration && <BoxIllustration aria-hidden="true">{illustration}</BoxIllustration>}
 
       <Content>
         {titleAsTextDisplay ? (
-          <Text.Display2>{title}</Text.Display2>
+          <Text.Display2 as="h1" aria-label={title} tabIndex={0}>{title}</Text.Display2>
```
Comment (nataliosilveiraext) [inline packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx:80]: tabindex="0" is not necessary here, the screen reader will read it correctly

File: packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx [hunk 2]
```diff
@@ -73,15 +73,15 @@ function ActionRequirement(props) {
...
         ) : (
-          <Title>{title}</Title>
+          <Title aria-label={title} tabIndex={0}>{title}</Title>
         )}
-        <Text.Body1 mt="small" color="deep">
+        <Text.Body1 mt="small" color="deep" aria-label={description} tabIndex={0}>
```
Comment (nataliosilveiraext) [inline packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx:84]: tabindex="0" is not necessary here, the screen reader will read it correctly

**Review by nataliosilveiraext — CHANGES_REQUESTED** [review]

**Review by rodrigosousa1 — APPROVED** [review]

**Review by JVCDomingues — APPROVED** [review]

**Review by diogoverde — APPROVED** [review]

File: packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx [hunk 3]
```diff
@@ -73,15 +73,18 @@ function ActionRequirement(props) {
...
-          <Text.Display2>{title}</Text.Display2>
+          <Text.Display2 as="h1" aria-label={title}>
+            {title}
+          </Text.Display2>
         ) : (
-          <Title>{title}</Title>
+          <Title aria-label={title}>{title}</Title>
         )}
-        <Text.Body1 mt="small" color="deep">
+        <Text.Body1 mt="small" color="deep" aria-label={description}>
```
Comment (leonardo-gympass) [inline packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx:87]: We don't need to add an `aria-label ` with the same value as children, the value of children will already be detected by assistive technologies.

File: packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx [hunk 4]
```diff
@@ -73,15 +73,18 @@ function ActionRequirement(props) {
   defineCompoundComponents();
   return (
     <StyledActionRequirement {...props}>
-      {illustration && <BoxIllustration>{illustration}</BoxIllustration>}
-
+      {illustration && (
+        <BoxIllustration aria-hidden="true">{illustration}</BoxIllustration>
```
Comment (leonardo-gympass) [inline packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx:77]:
```suggestion
        <BoxIllustration aria-hidden>{illustration}</BoxIllustration>
```

File: packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx (hunk 1 above)
Comment (mavericardo) [reply packages/yoga/src/ActionRequirement/web/ActionRequirement.jsx]: Done

**Review by leonardo-gympass — APPROVED** [review]

**Review by flavia-moraes — APPROVED** [review]

**Review by nataliosilveiraext — APPROVED** [review]

**Pull request merged**